security-force-monitor/sfm-cms
# Disable sources in downloads

## Overview

Disable the option to include sources in downloads. See the notes for some background on why we're making this change.

Connects #661 

### Notes

It turns out the queries that we use to create materialized views for downloads require too much disk space to run on the production database. (Disk space is required for temporary tables that are created to handle joins at query time.) We initially ran into this problem with the smaller subset of Mexico data and were able to fix it by splitting the query into subqueries such that we could perform filtering in a separate subquery and then join it to the main query; this worked for Mexico, but did not scale up to production. To test a fix locally, I downloaded a dump of the production database and restored it locally.

The fact that removing sources as done in this PR fixes the problem and reduces the amount of time it takes to run the queries from hours to a couple of seconds indicates to me that the highly normalized nature of the sources tables is the root problem here. Since each source attribute represents a many-to-many relationship and has its own table, including sources not only doubles the joins, but greatly increases the complexity of the query. But figuring out exactly how to join them properly will take somewhere between 4-8 hours and consultation with Forest, so for now we're going to disable the feature.

## Testing Instructions

* Run `docker-compose run --rm app ./manage.py make_materialized_views` and confirm that all the queries run successfully
* Navigate to http://localhost:8000/en/download and make sure you can download each download type for one given geography (I tested Mexico)
